### PR TITLE
Add test cases in sampler for WebGL 2

### PR DESCRIPTION
**Pull request by legendlee1314**
Test targets are getSamplerParameter and samplerParameter[if].
Checking the pnames and if the default and setting params are correct.
Will add params checking cases later.

### CODE REVIEW

**Comment by legendlee1314** [comment]
@kenrussell , @zhenyao , @toji , could you please review this? Thanks!

File: sdk/tests/conformance2/samplers/samplers.html [hunk 1]
```diff
@@ -126,6 +133,72 @@
...
+    testCases = [
+      { pname: gl.TEXTURE_WRAP_S,
+        param: gl.REPEAT,
+        expectedError: gl.NO_ERROR },
+      { pname: gl.TEXTURE_MIN_FILTER,
+        param: gl.LINEAR,
+        expectedError: gl.NO_ERROR },
+      { pname: gl.TEXTURE_COMPARE_MODE,
+        param: gl.NONE,
+        expectedError: gl.NO_ERROR },
```
Comment (zhenyao) [inline sdk/tests/conformance2/samplers/samplers.html]: Could you please also test valid pname TEXTURE_WRAP_R, TEXTURE_WRAP_T, TEXTURE_MAG_FILTER, TEXTURE_COMPARE_FUNC? The list is short, so it's good to verify them all.

Comment (legendlee1314) [inline sdk/tests/conformance2/samplers/samplers.html]: Yes, updated to have all valid pnames. Moreover, add tests for the default value of them.

**Comment by zhenyao** [comment]
Great!

**Pull request merged**